code/datums/traits.dm

```diff
@@ -1334,6 +1334,48 @@ TYPEINFO(/datum/trait/partyanimal)
 	category = list("species", "nopug", "nohair")
 	mutantRace = /datum/mutantrace/pug
 
+/datum/trait/random_species
+	name = "Random Species"
+	icon_state = "randomspecies"
+	desc = "You feel like something's different today, but you can't quite put your finger/tail/hoof/antennae on it."
+	id = "random_species"
+	points = -1
+	category = list("species", "infrared", "cloner_stuff", "nohair", "hemophilia")
+
+	onAdd(mob/owner)
+		if (ishuman(owner))
+			var/mob/living/carbon/human/H = owner
+			var/list/datum/mutantrace/default_species = list(
+						/datum/mutantrace/lizard,
+						/datum/mutantrace/cow,
+						/datum/mutantrace/skeleton,
+						/datum/mutantrace/roach,
+					)
+			var/datum/mutantrace/new_mutantrace_type = null
+			if (prob(1) && prob(1)) // 0.01% chance of a non-dna mutagen banned mutrace that *isn't* a -1 point mutrace
+				var/max_iterations = 50 // safety
+				var/iteration = 0
+				while ( \
+					isnull(new_mutantrace_type) || \
+					initial(new_mutantrace_type:dna_mutagen_banned) || \
+					(new_mutantrace_type in default_species) || \
+					(iteration < max_iterations) \
+				)
+					iteration += 1
+					new_mutantrace_type = pick(concrete_typesof(/datum/mutantrace))
```

Comment:
consider `filtered_concrete_typesof()`

Reply:
implemented here and where i copied the pattern from in cloner defects